azure/src/test/java/org/apache/iceberg/azure/AzurePropertiesTest.java

```diff
@@ -29,12 +34,50 @@
 import com.azure.core.credential.TokenCredential;
 import com.azure.storage.common.StorageSharedKeyCredential;
 import com.azure.storage.file.datalake.DataLakeFileSystemClientBuilder;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
 import org.assertj.core.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 public class AzurePropertiesTest {
 
+  @Test
+  public void testSerializable() {
+    AzureProperties props =
+        new AzureProperties(
+            ImmutableMap.<String, String>builder()
+                .put(ADLS_SAS_TOKEN_PREFIX + "foo", "bar")
+                .put(ADLS_CONNECTION_STRING_PREFIX + "foo", "bar")
+                .put(ADLS_READ_BLOCK_SIZE, "42")
+                .put(ADLS_WRITE_BLOCK_SIZE, "42")
+                .put(ADLS_SHARED_KEY_ACCOUNT_NAME, "me")
+                .put(ADLS_SHARED_KEY_ACCOUNT_KEY, "secret")
+                .build());
+
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    assertThatCode(
+            () -> {
+              try (ObjectOutputStream oos = new ObjectOutputStream(out)) {
+                oos.writeObject(props);
+              }
+            })
+        .doesNotThrowAnyException();
+
+    byte[] serialized = out.toByteArray();
```

Comment:
it would be good to actually create an instance of `AzureProperties` and verify that this matches `props`

Reply:
something like
```
AzureProperties props =
        new AzureProperties(
            ImmutableMap.<String, String>builder()
                .put(ADLS_SAS_TOKEN_PREFIX + "foo", "bar")
                .put(ADLS_CONNECTION_STRING_PREFIX + "foo", "bar")
                .put(ADLS_READ_BLOCK_SIZE, "42")
                .put(ADLS_WRITE_BLOCK_SIZE, "42")
                .put(ADLS_SHARED_KEY_ACCOUNT_NAME, "me")
                .put(ADLS_SHARED_KEY_ACCOUNT_KEY, "secret")
                .build());

    AzureProperties serdedProps = TestHelpers.roundTripSerialize(props);
    assertThat(serdedProps.adlsReadBlockSize()).isEqualTo(props.adlsReadBlockSize());
    assertThat(serdedProps.adlsWriteBlockSize()).isEqualTo(props.adlsWriteBlockSize());
    assertThat(serdedProps.adlsSasTokens()).isEqualTo(props.adlsSasTokens());
    assertThat(serdedProps.namedKeyCreds()).isEqualTo(props.namedKeyCreds());
    assertThat(serdedProps.adlsConnectionStrings()).isEqualTo(props.adlsConnectionStrings());
```